plugin/environment.go

```diff
@@ -428,14 +435,11 @@ func (env *Environment) RunMultiPluginHook(hookRunnerFunc func(hooks Hooks) bool
 	env.registeredPlugins.Range(func(key, value interface{}) bool {
 		rp := value.(*registeredPlugin)
 
-		if rp.supervisor == nil || !rp.supervisor.Implements(hookId) {
+		if rp.supervisor == nil || !rp.supervisor.Implements(hookId) || !env.IsActive(rp.BundleInfo.Manifest.Id) {
```

Comment:
I seem to recall the earlier code maintaining a set of `activePlugins` and a set of `registeredPlugins`. It doesn't look like there's a lot of overhead here, but I still wonder if we should frontload this work instead of repeating it for every hook (i.e. "post-level frequency")

Reply:
@lieut-data 
by frontloading this work, do you mean to store `activePlugins` in the `Environment` struct? along side of `registeredPlugins` or compute it here inside this method?

Don't think precomputing `active` plugins inside this method will make much difference, as in both cases we'll have to go through `registeredPlugins` and figure out `active` plugins. The only gain that's possible here is if we keep track of `active` plugins separately along with `registeredPlugins` in `Environment`.
Thoughts?

Reply:
IIRC, I was leaning towards storing `activePlugins` alongside `registeredPlugins` in `Environment`, but I'm just using the force here on performance concerns. Looking at what `IsActive` actually does, I don't see a specific concern here, so I'm fine leaving as-is.

Reply:
Reading this [article](https://medium.com/@deckarep/the-new-kid-in-town-gos-sync-map-de24a6bf7c2c) and watching this [video](https://www.youtube.com/watch?v=C1EtfDnsdDs) from the person who wrote `sync.Map`, looks like it works best for a high-read scenarios, so what we have here should be fine performance wise. 

Thinking it through even if we have `activePlugins`, we'll still be going through `.Load` or `.Rage` which has some sort of internal ready-only locking, so in the end it might not save us anything, but we won't know till we measure the overall performance of all these hook invocations :)